pandas/io/formats/format.py

```diff
@@ -77,6 +77,16 @@
     List[Callable], Tuple[Callable, ...], Mapping[Union[str, int], Callable]
 ]
 FloatFormatType = Union[str, Callable, "EngFormatter"]
+ColspaceType = Union[
+    Mapping[Union[str, int], Union[str, int]],
+]
+ColspaceArgType = Union[
+    str,
+    int,
+    List[Union[str, int]],
+    Tuple[Union[str, int], ...],
+    Mapping[Union[str, int], Union[str, int]],
```

Comment:
```suggestion
    Mapping[Label, Union[str, int]],
```

Minor but we allow more than just str / int objects as column labels. You can import this from `pandas._typing`